pkg/driverbuilder/builder/amazonlinux.go

```diff
@@ -62,26 +62,26 @@ type amazonlinuxTemplateData struct {
 	KernelDownloadURLs []string
 }
 
-func (a amazonlinux) Name() string {
+func (a *amazonlinux) Name() string {
 	return TargetTypeAmazonLinux.String()
 }
 
-func (a amazonlinux) TemplateScript() string {
+func (a *amazonlinux) TemplateScript() string {
 	return amazonlinuxTemplate
 }
 
-func (a amazonlinux) URLs(_ Config, kr kernelrelease.KernelRelease) ([]string, error) {
+func (a *amazonlinux) URLs(_ Config, kr kernelrelease.KernelRelease) ([]string, error) {
 	return fetchAmazonLinuxPackagesURLs(a, kr)
 }
 
-func (a amazonlinux) TemplateData(c Config, kr kernelrelease.KernelRelease, urls []string) interface{} {
+func (a *amazonlinux) TemplateData(c Config, kr kernelrelease.KernelRelease, urls []string) interface{} {
 	return amazonlinuxTemplateData{
-		commonTemplateData: c.toTemplateData(),
+		commonTemplateData: c.toTemplateData(a, kr),
```

Comment:
`c.toTemplateData(a, kr)` now internally calls `c.setGCCVersion`, that's why it needs both the kernelrelease and the builder now.